### DISCUSSION TITLE
STILL can't get vvveb cms to install and work correctly. Really frustrated here.

### DISCUSSION THREAD

**Discussion by onjaysantana** [givanz/Vvveb#134] [General]
Sorry to post basically my same question, but I've tried what I could find on this site. I honestly think I must REALLY misunderstand something basic or what everyone sees as obvious.

I'm trying to set up vvveb (the cms, not just the editor) on a shared cPanel Linux account (ISP is InterServer). I tried setting it up on a Windows server with the same ISP, but had essentially the same problem there. In fact, I opened a Linux account just to try to get vvveb installed. It really seems like what I'm looking for. I tried the demo.

I am partly confused on just where the files should be placed in the file system. I created a subdomain for vvveb and set up the database, which exists and is populated with a bunch of stuff. When I go to the URL (you can try it: vvveb1.johnsantana.work I get lots of errors that look like issues with the database. Beyond what I know about the program.  Here are the first few errors thrown:

Error

FUNCTION johnsant_vvveblinux1.JSON_ARRAYAGG does not exist
SELECT  pd.*,products.*, products.product_id as array_key
						 
				,(SELECT JSON_ARRAYAGG( JSON_OBJECT('id', pi.product_image_id, 'image', pi.image) ) 
				FROM product_image as pi WHERE pi.product_id = products.product_id GROUP BY pi.product_id) as images

Sure looks like problems grabbing stuff from the database. NOTE: If you point your browser to the public folder (vvveb1.johnsantana.work/public, the an unformatted page comes up with the ability to click on the links to other pages like About, and the pages come up. But not the home page. 

HOWEVER: Even when I managed to get the program partly working on the windows server, the page files were reported as missing, even though they were in the page list. 

ALSO: Using just the vvveb.js editor, I get problems. Like after selecting an image on a page, pressing the set image button does nothing. Also, sometimes I've not been able to save work, though the program seems to have all permissions.

Look, I know this must seem like I'm a real know-nothing. I suppose it's true with regards to setting this up. Access to the command line is not allowed with this hosting account, but it should not be needed.

I may be way off base, but it looks like I'm not getting the latest.zip file uploaded to the folder the program likes--it does seem pretty particular about that. The fact that the public folder gets the ability to bring up a page suggests that.

Sure would like to get this to work. I've had it with Wordpress. I've tried a couple of others, but they are either slow or inflexible for what I need.

Please help if you can. I'm willing to pay a small fee to someone who get get it working for me.

Best,
JS

**Comment by givanz** [comment]
Hi

It looks that you have an old version of Mysql/MariaDB because JSON_ARRAYAGG is missing.

JSON_ARRAYAGG was added in MySQL version 5.7.22 and in MariaDB version 10.5.0

You need to upgrade your database or reinstall using SQLite instead of MySql.

You can delete config/db.php to reset db config and install page will show to choose SQLite.

Let me know if you still have issues after installing with SQLite.

**Comment by onjaysantana** [comment]
Hi,
Thanks SO MUCH for your help. I'll see what's up at my ISP (InterServer).
They always seem behind on versions of everything.

Best,
JS

**Comment by onjaysantana** [comment]
Just a very quick follow up. Interserver's db is MariaDb 10.3.3. I'll lean
on them. Meanwhile, I'll try my other provider.
JS

**Comment by onjaysantana** [comment]
While waiting for my Linux account ISP to update MariaDb, I have installed
vvVeb on my Windows hosting account (uses IIS).

The landing page displays but the other pages report error 404. I
understand that an Apache setting (rewrite?) can affect that. But what
about Windows IIS?

Also, if I point the browser to the public folder, the page comes up, but
not formatted like this:


[image: image.png]

Does this tell you anything?

I repeat that I am very sorry for the amount of help I've needed, but I
would like to get this going.If you could get back to me on this, I sure
would appreciate it.

Best,
JS

**Comment by givanz** [comment]
Apache shows 404 not found if either mod rewrite module is not enabled or if AllowOverride is not set to load the rules from the .htaccess file.

Mod rewrite can be enabled from command line with 

```bash
sudo a2enmod rewrite
```

.htaccess can be enabled by adding `AllowOverride All` in apache .conf file.

```apache2
<Directory "/var/html/vvveb/">
 Require all granted
 AllowOverride All
</Directory>
```

I never used IIS and I don't know how rewrite is enabled.

For windows probably the easieast way is to use docker https://docs.docker.com/desktop/install/windows-install/ and install the Vvveb docker image https://hub.docker.com/r/vvveb/vvvebcms

**Comment by onjaysantana** [comment]
Again, thanks for all your help. It looks like I'm almost there. At least
major progress.

My ISP updated MariaDb and I'm able to log in and most everything looks and
works as expected.

EXCEPT, when I try to SAVE changes I get an error dialog with the
message " Error
saving!"
I also get the error on creating a new page, BUT, it seems like I can
RENAME a page.

Right now, if got permissions set for everything for everyone (777), which
I don't think should have to be.

Also, I see a comment in the docs somewhere that I may need to run
editor.php, or is that just for vvvebjs?

So, I'm still not there, and I'm hoping you can point me in the right
direction.

Thanks
JS

**Comment by givanz** [comment]
Common issues with page save include low value for PHP `post_max_size` you can check the current value in admin > tools > system info https://demo.vvveb.com/admin/index.php?module=tools/systeminfo

Or a low value for apache mod security `SecRequestBodyLimit`

You can check php and apache error logs to see if there is a limit or permission denied error for page save.

PHP error logs can be viewed also from admin > tools > error log https://demo.vvveb.com/admin/index.php?module=tools/errorlog

mod security error should look like this https://github.com/givanz/Vvveb/discussions/131#discussioncomment-9748453 in apache logs

I made some changes to page save code in the last update to show the return status on error for more details.

Accessing editor.php directly is only needed for VvvebJs.

**Comment by onjaysantana** [comment]
Hi,
Thanks again. You have been very generous in your replies. I made a couple
of tweeks to the php settings you mentioned and dropped back to php 8.1.
Could you take a quick look at these settings from the tools in vvVeb and
see if there is something obvious that needs to be fixed? I'm knowledgeable
in many areas, but this isn't one of them, so I do need help.

So, if you're willing to take a look, here's the data:

General

      Vvveb version
      0.0.7

      Admin path
      /admin/

      PHP version
      8.1.28 | litespeed

      Server
      LiteSpeed

      OS version
      Linux webhosting2053.is.cc 4.18.0-513.18.1.lve.2.el8.x86_64 #1 SMP
Sat Mar 30 15:36:11 UTC 2024 x86_64

      Database driver & version
      mysqli | Client library version: mysqlnd 8.1.28 | Server version:
10.5.25-MariaDB-cll-lve

      PHP time limit
      60

      PHP memory limit
      256M

      Max input time
      60

      Upload max filesize
      200M

      PHP post max size
      200M

      Extensions
      Core date libxml openssl pcre sqlite3 zlib bz2 calendar ctype curl
hash filter ftp gettext json iconv SPL pcntl readline Reflection session
standard mbstring shmop SimpleXML tokenizer xml litespeed i360 bcmath dba
dom enchant fileinfo gd imap intl ldap exif mysqlnd mysqli odbc PDO
pdo_mysql PDO_ODBC pdo_pgsql pdo_sqlite pgsql Phar posix pspell snmp soap
sockets sysvmsg sysvsem sysvshm tidy xmlreader xmlwriter xsl zip

      Page cache
      enabled

      Object cache
      APCu

      Email Driver
      mail

      Session Driver
      php

      Debug
      enabled

      Sql changes check
      enabled

      Image library
      gd

Server

      Document root
      /home/johnsant/vvveb1.johnsantana.work

      Public path
      /public/

      Path
      /usr/local/bin:/bin:/usr/bin

      Http accept

text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8

      Http accept encoding
      gzip, deflate, br, zstd

      Http accept language
      en-US,en;q=0.5

      Http cookie
      PHPSESSID=rzJl%2CUQqCtWQD6yjiPuTzQIyNN-M55jMtfQ9mMiZrZkkAHdQ; user=1

      Http host
      vvveb1.johnsantana.work

      Http referer

https://vvveb1.johnsantana.work/admin/index.php?module=tools/update&action=check&type=core

      Http user agent
      Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:127.0) Gecko/20100101
Firefox/127.0

      Http alt used
      vvveb1.johnsantana.work

      Http upgrade insecure requests
      1

      Http sec fetch dest
      document

      Http sec fetch mode
      navigate

      Http sec fetch site
      same-origin

      Http sec fetch user
      ?1

      Http priority
      u=1

      Remote addr
      172.115.51.140

      Remote port
      65118

      Server addr
      66.45.238.235

      Server name
      vvveb1.johnsantana.work

      Server admin
      ***@***.***

      Server port
      443

      Request scheme
      https

      Request uri
      /admin/index.php?module=tools/systeminfo

      Https
      on

      X spdy
      HTTP3

      Ssl protocol
      QUIC

      Ssl cipher
      TLS_AES_128_GCM_SHA256

      Ssl cipher usekeysize
      128

      Ssl cipher algkeysize
      128

      Script filename
      /home/johnsant/vvveb1.johnsantana.work/admin/index.php

      Query string
      module=tools/systeminfo

      Script uri
      https://vvveb1.johnsantana.work/admin/index.php

      Script url
      /admin/index.php

      Script name
      /admin/index.php

      Server protocol
      HTTP/1.1

      Server software
      LiteSpeed

      Request method
      GET

      X lscache
      on

      Php self
      /admin/index.php

      Request time float

1718648014.792967

      Request time

1718648014

**Comment by givanz** [comment]
`post_max_size` has a high value, this is not the cause of the error.

Can you also please check apache error log for mod security errors and paste the last errors here? 

If you are using cpanel, you can use this guide https://www.inmotionhosting.com/support/edu/cpanel/how-to-view-cpanel-error-logs/

You can also check the php error log from admin > tools > error log or if it's empty from cpanel dashboard.

**Comment by onjaysantana** [comment]
Thanks. I looked into logs I could find, and located this list. Looks like
something is not being found. What do you think?

2024-06-17 14:12:56.399920 [INFO] [3016139] [T0] [172.115.51.140:49717
-H3:769D12-316#APVH_vvveb1.johnsantana.work:443] File not found
[/home/johnsant/vvveb1.johnsantana.work/403.shtml]
2024-06-17 14:08:23.485298 [INFO] [2903660] [T0] [172.115.51.140:59331
-H3:412BEAA6D6FC0AF2-576#APVH_vvveb1.johnsantana.work:443] File not found
[/home/johnsant/vvveb1.johnsantana.work/403.shtml]
2024-06-17 14:07:06.699628 [INFO] [2903660] [T0] [172.115.51.140:49963
-H3:184E06D9C6C2288E-184#APVH_vvveb1.johnsantana.work:443] File not found
[/home/johnsant/vvveb1.johnsantana.work/403.shtml]
2024-06-17 14:05:48.840193 [INFO] [2903660] [T0] [172.115.51.140:52586
-H3:024CF340CE6E5596-576#APVH_vvveb1.johnsantana.work:443] File not found
[/home/johnsant/vvveb1.johnsantana.work/403.shtml]
2024-06-17 13:57:55.504195 [INFO] [2903658] [T0] [172.115.51.140:50327
-H3:9B15DA07BBA576B1-188#APVH_vvveb1.johnsantana.work:443] File not found
[/home/johnsant/vvveb1.johnsantana.work/403.shtml]
2024-06-17 13:56:51.625043 [INFO] [2903658] [T0] [172.115.51.140:58521
-H3:2906C7A3A1B4921F-600#APVH_vvveb1.johnsantana.work:443] File not found
[/home/johnsant/vvveb1.johnsantana.work/403.shtml]
2024-06-17 13:54:56.457938 [INFO] [2903658] [T0] [172.115.51.140:65494
-H3:DABF5A0527B4A2E2-460#APVH_vvveb1.johnsantana.work:443] File not found
[/home/johnsant/vvveb1.johnsantana.work/403.shtml]
2024-06-17 13:53:15.398383 [INFO] [2903658] [T0] [172.115.51.140:55365
-H3:75FB2D-696#APVH_vvveb1.johnsantana.work:443] File not found
[/home/johnsant/vvveb1.johnsantana.work/403.shtml]
2024-06-17 13:52:55.755306 [INFO] [2903660] [T0] [172.115.51.140:54452
-H3:6BC149-0#APVH_vvveb1.johnsantana.work:443] File not found
[/home/johnsant/vvveb1.johnsantana.work/403.shtml]
2024-06-17 13:50:45.929526 [INFO] [2903658] [T0] [172.115.51.140:62182
:HTTP2-365#APVH_vvveb1.johnsantana.work:443] File not found [/home/johnsant/
vvveb1.johnsantana.work/403.shtml]
2024-06-17 13:50:18.525633 [INFO] [2903660] [T0] [172.115.51.140:65207
-H3:80739C-12#APVH_vvveb1.johnsantana.work:443] File not found
[/home/johnsant/vvveb1.johnsantana.work/403.shtml]
2024-06-17 13:50:06.715878 [INFO] [2903660] [T0] [172.115.51.140:65207
-H3:80739C-4#APVH_vvveb1.johnsantana.work:443] File not found
[/home/johnsant/vvveb1.johnsantana.work/403.shtml]
2024-06-17 13:46:12.667203 [INFO] [2889268] [T0] [172.115.51.140:63872
-H3:58C3D3-28#APVH_vvveb1.johnsantana.work:443] File not found
[/home/johnsant/vvveb1.johnsantana.work/403.shtml]

**Comment by onjaysantana** [comment]
Hi,

Hold the phone! I'm on to something. I found that turning of ModSecurity
seems to have fixed most issues, though I'm still checking areas where I
had problems before. Pages are saving (so far), but I need to dig a little
deeper.

Maybe you should not worry about this for now, though it would be good to
know how to use ModSecurty. Settings, etc.

Best,
John

**Comment by givanz** [comment]
Hi

Thanks for feedback.

Mod security default value for post body limit is very low.

If mod security is enabled then there is a high change save will not work without adding an exception for page save (admin/controller/editor/editor.php) or increasing the limit.

Page save needs a bigger value because it send the whole page html through post and page size is usually higher than the mod sec post body limit.

**Comment by winup168** [comment]
@onjaysantana 
Here's your description translated into English:

---

I am a new user of VVVeb CMS from China. Due to differences in language capabilities, installing VVVeb has been quite challenging. With the help of ChatGPT for translation and guidance, I encountered persistent errors when trying to install it with SQLite. Despite ChatGPT's efforts to troubleshoot, the issue was not resolved after several hours.

Feeling exhausted from the lengthy process, I took a break and decided to start from scratch with a different approach. I upgraded to MySQL version 5.7.44, then removed all previous attempts, including recreating the database. Surprisingly, the installation was successful this time!

However, the site language cannot be localized. Although I have installed the CN language pack and it shows as successful, the Chinese interface still does not display. This issue is still a work in progress.

**Comment by givanz** [comment]
Sorry to hear about the sqlite issues, can you provide the php and sqlite version you used? maybe I can reproduce the errors, some older sqlite version lack some features like full text search.

For Chinese language currently only zh_CN 🇨🇳 简体中文 (Chinese) is available.

I updated and installed the language pack on the demo https://demo.vvveb.com/admin/index.php?module=localization/language&language_id=4 and it's working.

Can you please try again? You can delete the `/vvveb/locale/zh_CN` folder to remove the old language pack.

Some possible issues can be related to file write permissions, please check if `/vvveb/locale/zh_CN/LC_MESSAGES/vvveb.po` and `/vvveb/locale/zh_CN/LC_MESSAGES/vvveb.mo` files exists after installing the language pack.

If only `/vvveb/locale/zh_CN/LC_MESSAGES/vvveb.mo` is missing then there might be a compilation error, you can manually compile to mo with `msgfmt vvveb.po` or open vvveb.po with something like poeditor.

You can also try to restart your http server (php-fpm/nginx/apache) sometimes php caches gettext translations.
